text/048-packaging-coq-native.md

```diff
@@ -0,0 +1,180 @@
+| yes | **ondemand** | ondemand (default if omitted) | temporary | none |
+| yes | **ondemand** | no | none | none |
+| no | no | yes | none | none |
+| no | no | ondemand | none | none |
+| no | no | no | none | none |
+
+Note also that (for each version of coq), **the stdlib is only precompiled with `./configure -native-compiler yes`**. It is not precompiled with `./configure -native-compiler ondemand` for `coq >= 8.13`.
+  
+[PR coq/coq#13352](https://github.com/coq/coq/pull/13352) implements this item.
+
+## Regarding item 3:
+
+Assuming we want to use `native_compute` on Coq < 8.13 with `coq-mathcomp-ssreflect` (or any library built with `coq_makefile`), the `opam` file has to be extended like this (see e.g. [Coq's refman](https://coq.github.io/doc/master/refman/practical-tools/utilities.html?highlight=coq_makefile#precompiling-for-native-compute)): 
+
+```diff
+diff --git a/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.10.0/opam b/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.10.0/opam
+--- a/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.10.0/opam
++++ b/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.1.10.0/opam
+@@ -6,7 +6,8 @@
+ dev-repo: "git+https://github.com/math-comp/math-comp.git"
+ license: "CeCILL-B"
+ 
+-build: [ make "-C" "mathcomp/ssreflect" "-j" "%{jobs}%" ]
++build: [ make "-C" "mathcomp/ssreflect" "-j" "%{jobs}%" "COQEXTRAFLAGS+=-native-compiler yes" {coq-native:installed}]
+ install: [ make "-C" "mathcomp/ssreflect" "install" ]
+ depends: [ "coq" { (>= "8.7" & < "8.12~") } ]
+```
+
+Doing this, one can notice that the appropriate `.coq-native` directory has successfully been installed along the `.vo` of the library: 
+
+```bash
+find "$(opam config var lib)" -name ".coq-native" | grep ssreflect
+```
+
+# Note to library developers and package maintainers
+
+To fully benefit from `native_compute` with a library using Coq 8.5+, some (`opam`) packaging update may be required for this library's dependencies:
+
+- for `coq_makefile`: no change required for Coq 8.13+; otherwise follow **item 3** above (related to Section [Precompiling for `native_compute`](https://coq.github.io/doc/master/refman/practical-tools/utilities.html?highlight=coq_makefile#precompiling-for-native-compute) in Coq refman);
+- for `dune / coq.theory`: this will require Coq 8.12.1+ and a version of `dune` implementing [PR ocaml/dune#3210](https://github.com/ocaml/dune/pull/3210) (cf. [this comment](https://github.com/coq/ceps/pull/48#issuecomment-727020253) by **@ejgallego**); otherwise (without native support), Dune will just ignore the native parts and no `./coq-native/*.cmxs` file will be installed.
+
+Note that these changes could be performed directly in the existing packages gathered in the [coq/opam-coq-archive](https://github.com/coq/opam-coq-archive) repository.
```

Comment:
Please make this section more visible, at least by [searching "native" in refman](https://coq.inria.fr/distrib/current/refman/search.html?q=native&check_keywords=yes&area=default).

I found here by searching "native" on Discourse and following the link in [this post](https://coq.discourse.group/t/about-the-native-compiler-flag-and-the-coq-native-opam-package-to-be-released-before-coq-8-13-beta1/1147?u=lys), which doesn't seem so intuitive.

Reply:
Actually since Dune 3.0 we could just make native work automatically, I'm gonna open a dune issue.

Reply:
Please emphasize that Coq stanza should also be upgraded to `(coq 0.3)`. I figured it out by reading between the lines in ocaml/dune#3210.